crates/rust-analyzer/src/config.rs

```diff
@@ -651,23 +654,304 @@ pub struct Config {
     workspace_roots: Vec<AbsPathBuf>,
     caps: lsp_types::ClientCapabilities,
     root_path: AbsPathBuf,
-    detached_files: Vec<AbsPathBuf>,
```

Comment:
Let's keep that here, it's a special thing so we should really handle it specially. (it being encoded in the config is technically wrong)